File: dev-docs/snippets/analytics/log-snapshot.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Emit a "snapshot" analytics event with arbitrary context.
// JS-only — the Swift client has no equivalent.
export function logSnapshot(client: JsBaoClient) {
  // #region example
  client.analytics.logSnapshot({ screen: "settings", theme: "dark" });
  // #endregion example
}

File: dev-docs/snippets/analytics/log-snapshot.swift
import JsBaoClient

// Emit a "snapshot" analytics event (`action: "_snapshot"`, `feature: "_state"`)
// with arbitrary context. No-ops when there is no signed-in user.
func logSnapshot(client: JsBaoClient) {
  // #region example
  client.analytics.logSnapshot(context: ["screen": "settings", "theme": "dark"])
  // #endregion example
}
